### PR TITLE
Introduce set function into storage maps

### PR DESCRIPTION
**Pull request by gavofyork**
Storage value already had a `set` API for "setting" the value from a query instance (which if it's an `OptionQuery` would actually remove the value if it's a `None`). Maps did not have this, but now they do 🎉

### CODE REVIEW

File: frame/support/src/storage/generator/double_map.rs [hunk 1]
```diff
@@ -151,6 +151,13 @@ where
 		unhashed::get(&Self::storage_double_map_final_key(k1, k2)).ok_or(())
 	}
 
+	fn set<KArg1: EncodeLike<K1>, KArg2: EncodeLike<K2>>(k1: KArg1, k2: KArg2, q: Self::Query) {
```
Comment (cheme) [inline frame/support/src/storage/generator/double_map.rs:154]:
```suggestion
	fn set<KArg1, KArg2>(k1: KArg1, k2: KArg2, q: Self::Query)
	where
		KArg1: EncodeLike<K1>,
		KArg2: EncodeLike<K2>,
	{
```

**Review by KiChjang — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Comment by gavofyork** [comment]
Test failure unrelated.

File: frame/support/src/storage/types/double_map.rs [hunk 2]
```diff
@@ -203,6 +203,15 @@ where
 		<Self as crate::storage::StorageDoubleMap<Key1, Key2, Value>>::try_get(k1, k2)
 	}
 
+	/// Store or remove the value to be associated with `key` so that `get` returns the `query`.
+	pub fn set<KArg1: EncodeLike<Key1>, KArg2: EncodeLike<Key2>>(
+		k1: KArg1,
+		k2: KArg2,
+		q: QueryKind::Query,
+	) {
```
Comment (cheme) [inline frame/support/src/storage/types/double_map.rs:211]:
```suggestion
pub fn set<KArg1, KArg2>(k1: KArg1, k2: KArg2, q: Self::Query)
	where
		KArg1: EncodeLike<K1>,
		KArg2: EncodeLike<K2>,
	{
```

**Review by cheme — APPROVED** [review]
LGTM, sometime calls to function like `storage_double_map_final_key` can be extracted from `insert` and `remove`, but I don't think it would make a big difference. Code in `mutate` (in double_map and maybe at different places) also can use the new `set` function.

**Pull request merged**